Covid19Radar/Covid19Radar/Services/TestNativeImplementation.cs

```diff
@@ -4,79 +4,198 @@
+                    return (
+                        new ushort[] {0, 0, 0, 0, 0, // ExposureDetectionSummary
+                        });
+                case PresetDataType.TwoLowRiskMatches:
+                default:
+                    return (
+                        new ushort[] {10, 2, 5, 0, 0, // ExposureDetectionSummary
+                                       10, 15, 65, 5, (ushort)RiskLevel.Medium, // ExposureInfo 1st (RiskLevel.Medium=4)
+                                       11,  5, 40, 3, (ushort)RiskLevel.Low, // ExposureInfo  2nd(RiskLevel.Low=2)
+                        });
+            }
+        }
+
+        private ushort[] CreatePresetData()
+        {
+            if (mockCommonUtils.IsDirectInputApi())
+            {
+                return mockCommonUtils.GetApiStrings().Select(x => Convert.ToUInt16(x)).ToArray();
+            }
+            return DataPreset(mockCommonUtils.GetApiDataType());
+
+        }
+
+        public Task<(ExposureDetectionSummary summary, Func<Task<IEnumerable<ExposureInfo>>> getInfo)> DetectExposuresAsync(IEnumerable<string> files)
+        {
+            /* ApiUrlBase trick for Debug_Mock
+            "10,2,5,0,0,10,15,65,5,4,11,5,40,3,2" -> direct input (the same with default)
+            "https://API_URL_BASE/api2" -> dataVer = 2
+            "https://API_URL_BASE/api" -> dataVer = 0 (default)
+            others -> dataVer is the number at the end of the sentence
+            */
+            var dataPreset = CreatePresetData();
+            int index = 0;
+            var summary = new ExposureDetectionSummary(dataPreset[index++], dataPreset[index++],
+                dataPreset[index++], new TimeSpan[dataPreset[index++]], dataPreset[index++]);
+            // c.f.: ExposureDetectionSummary(daysSinceLastExposure=dataPreset[0],matchedKeyCount=dataPreset[1],maximumRiskScore=dataPreset[2],attenuationDurations=new TimeSpan[dataPreset[3]],summationRiskScore=dataPreset[4])
+
+            Task<IEnumerable<ExposureInfo>> GetInfo()
+            {
+                var info = new List<ExposureInfo>();
+                while (index < dataPreset.Length)
+                {
+                    info.Add(new ExposureInfo(DateTime.UtcNow.AddDays(-dataPreset[index++]),
+                        TimeSpan.FromMinutes(dataPreset[index++]), dataPreset[index++],
+                        dataPreset[index++], (Xamarin.ExposureNotifications.RiskLevel)dataPreset[index++]));
+                    // c.f.: ExposureInfo(DateTime timestamp, TimeSpan duration, int attenuationValue, int totalRiskScore, RiskLevel riskLevel)
+                };
+                return Task.FromResult<IEnumerable<ExposureInfo>>(info);
+            }
+
+            return Task.FromResult<(ExposureDetectionSummary, Func<Task<IEnumerable<ExposureInfo>>>)>((summary, GetInfo));
+        }
+
+        static TemporaryExposureKey GenerateRandomKey(int daysAgo)
+        {
+            var buffer = new byte[16];
```

Comment:
元ファイルがこうなっていることを理解した上で、ここは`buffer`として使用しているわけではないので`keyData`とした方がわかりやすいかと思います。
また、`16`については定数`KEY_DATA_LENGTH`にしておいてください。

```suggestion
            var keyData = new byte[KEY_DATA_LENGTH];
```

`TemporaryExposureKey`の引数`rollingStart`と`rollingDuration`の値域が接触確認APIの仕様に合っていない気がしますが、これについては後で僕の方でfeature/debug_pageにPR送ろうかと思います。